web-marketplace/src/ledger.tsx

```diff
@@ -83,26 +83,21 @@ const getApi = async (
 
 export const LedgerProviderWithWallet: React.FC<React.PropsWithChildren<{}>> =
   ({ children }) => {
-    const { wallet, loaded } = useWallet();
+    const { wallet } = useWallet();
 
-    return (
-      <LedgerProvider wallet={wallet} walletLoaded={loaded}>
-        {children}
-      </LedgerProvider>
-    );
+    return <LedgerProvider wallet={wallet}>{children}</LedgerProvider>;
   };
 
 export const LedgerProvider: React.FC<
-  React.PropsWithChildren<{ wallet?: Wallet; walletLoaded: boolean }>
-> = ({ walletLoaded, wallet, children }) => {
+  React.PropsWithChildren<{ wallet?: Wallet }>
+> = ({ wallet, children }) => {
   const [api, setApi] = useState<RegenApi | undefined>(undefined);
   const [loading, setLoading] = useState<boolean>(false);
   const [error, setError] = useState<unknown>(undefined);
 
   useEffect(() => {
-    if (walletLoaded && !api)
-      getApi(setApi, setLoading, setError, wallet?.offlineSigner);
-  }, [api, setApi, setLoading, setError, wallet?.offlineSigner, walletLoaded]);
+    if (!api) getApi(setApi, setLoading, setError, wallet?.offlineSigner);
```

Comment:
this will cause another bug: since connecting to keplr wallet is done async, `wallet?.offlineSigner` will be undefined initially, which means we will instantiate `api` without the signer (even though there could be one)
then once `wallet?.offlineSigner` gets defined, we won't reset the `api` because it's already set (`if (!api)`)
meaning that you won't be able to sign transactions although your wallet is connected

I believe we could check in addition the following: `!!wallet?.offlineSigner && !api.msgClient` (`msgClient` is undefined if there's no signer: https://github.com/regen-network/regen-js/blob/v0.7.1/packages/api/src/api.ts#L66)

Reply:
@blushi see whether the latest updates sort this issue please.

> I believe we could check in addition the following..

Should we create a new issue to update this?

Reply:
just using the following instead does the same thing as the code updates you added:
```suggestion
    if (!api || (!!wallet?.offlineSigner && !api.msgClient)) getApi(setApi, setLoading, setError, wallet?.offlineSigner);
```
but it looks simpler since we don't need a new `useEffect` nor a new state variable